packages/frontend/editor-ui/src/features/credentials/components/CredentialEdit/CredentialEdit.vue

```diff
@@ -569,16 +573,35 @@ function onChangeSharedWith(sharedWithProjects: ProjectSharingData[]) {
 
 function onDataChange({ name, value }: IUpdateInformation) {
 	// skip update if new value matches the current
-	if (credentialData.value[name] === value) return;
+	const currentValue = get(credentialData.value, name);
+	if (currentValue === value) {
+		return;
+	}
 
 	hasUnsavedChanges.value = true;
 
 	const { oauthTokenData, ...credData } = credentialData.value;
-
-	credentialData.value = {
-		...credData,
-		[name]: value as CredentialInformation,
-	};
+	credentialData.value = deepCopy(credData);
+
+	if (value === undefined) {
+		// for fixed collections:
+		// if the value is undefined and the name (path) ends with a number
+		// we need to remove the item from the array
+		if (name.endsWith(']')) {
```

Comment:
Not sure but maybe it could simplify things:

In `NodeSettings.vue`, this helper function is used `updateNodeParameter` from `useNodeSettingsParameters`